### PR TITLE
Use Java NIO instead of File.

### PR DESCRIPTION
**Pull request by Articdive**
#### Description: 

Changes CommentedConfiguration and other areas of code to use Java NIO's "Path" instead of File. This removes certain methods from FileMgmt, I would argue reduces code complexity and in turn also improves the logging for certain error cases that we had usually just "returned false" with. In-theory this should also be a tiny bit faster on Java 11+ machines but it's not significant. 

Important: Did not touch the database for obvious reasons. Hence it still uses File.
____
#### New Nodes/Commands/ConfigOptions: 

Nope

____
#### Relevant Towny Issue ticket:

Nope

____
To Test:

Things I have tested:
- Outdated language files are still overriden (with a newer version).
- Fresh Install & Update
- Townyadmin reload
- Sister plugins depending on FileMgmt methods that were removed. (No references to FileMgmt methods were found in the newer TownyPlugins and TownyChat was tested and works fine)

- [x] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

File: src/com/palmergames/bukkit/towny/Towny.java [hunk 1]
```diff
@@ -168,7 +170,7 @@ public void onEnable() {
 			addMetricsCharts();
 		} catch (TownyInitException tie) {
 			addError(tie.getError());
-			getLogger().log(Level.SEVERE, tie.getMessage(), tie.getStackTrace());
+			getLogger().log(Level.SEVERE, tie.getMessage(), tie);
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/Towny.java:173]: Do we not want the stacktrace for problem solving? What is the tie going to show the log?

File: src/com/palmergames/bukkit/towny/Towny.java [hunk 2]
```diff
@@ -333,12 +333,12 @@ private void loadTownAndNationLevels() {
 	 * @since 0.97.0.24
 	 */
 	private boolean checkForLegacyDatabaseConfig() {
-		File file = getDataFolder().toPath().resolve("settings").resolve("config.yml").toFile();
+		Path configYMLPath = getDataFolder().toPath().resolve("settings").resolve("config.yml");
 		// Bail if the config doesn't exist at all yet.
-		if (!file.exists())
+		if (Files.exists(configYMLPath))
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/Towny.java:338]: This looks like backwards logic. We're supposed to bail on converting from a config that doesn't exist yet. This code is run before the config will be generated for the first time on a fresh install.

**Review by LlmDl — CHANGES_REQUESTED** [review]
Found a couple of small things and a rather large thing: the jar as it stands now does not pass a fresh install. Please test fresh install.

File: src/com/palmergames/bukkit/towny/Towny.java (hunk 1 above)
Comment (Articdive) [reply src/com/palmergames/bukkit/towny/Towny.java:173]: Using getStackTrace() sent it to a different method which thought these were log parameters, basically ignoring it.

File: src/com/palmergames/bukkit/towny/Towny.java (hunk 2 above)
Comment (Articdive) [reply src/com/palmergames/bukkit/towny/Towny.java:338]: Yeah good catch, fixed.

**Review by LlmDl — APPROVED** [review]

**Pull request merged**